src/ASM/LR/ASMu2D.C

```diff
@@ -1776,58 +1766,58 @@ bool ASMu2D::updateDirichlet (const std::map<int,RealFunc*>& func,
   std::map<int,RealFunc*>::const_iterator fit;
   std::map<int,VecFunc*>::const_iterator vfit;
   std::vector<DirichletEdge>::const_iterator dit;
-  std::vector<Ipair>::const_iterator nit;
-
+ 
   for (size_t i = 0; i < dirich.size(); i++)
   {
-    // Project the function onto the spline curve basis
-    Go::SplineCurve* dcrv = 0;
+    RealArray edgeControlpoints;
     if ((fit = func.find(dirich[i].code)) != func.end())
-      dcrv = SplineUtils::project(dirich[i].curve,*fit->second,time);
-    else if ((vfit = vfunc.find(dirich[i].code)) != vfunc.end())
-      dcrv = SplineUtils::project(dirich[i].curve,*vfit->second,nf,time);
+      edgeL2projection(dirich[i], *fit->second, edgeControlpoints, time);
     else
     {
       std::cerr <<" *** ASMu2D::updateDirichlet: Code "<< dirich[i].code
                 <<" is not associated with any function."<< std::endl;
       return false;
     }
-    if (!dcrv)
+    if (edgeControlpoints.empty())
     {
       std::cerr <<" *** ASMu2D::updateDirichlet: Projection failure."
                 << std::endl;
       return false;
     }
+ 
+    // Loop over the nodes of this boundary curve
+    for (size_t j = 0; j<dirich[i].MLGN.size(); j++)
+    {
+      if (dirich[i].MLGN[j] == -1) continue; // skip non-boundary nodes
```

Comment:
aha. missed this. no. just no. this needs to be a compressed index set for efficiency reasons.